### PR TITLE
Add support for leads

### PR DESCRIPTION
**Pull request by qzminski**
This pull request adds support for https://github.com/terminal42/contao-leads.

### CODE REVIEW

**Review by fritzmg — APPROVED** [review]

File: src/EventListener/LeadsListener.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    public function getDuplicateFields(array $allFields): array
+    {
+        static $duplicateFields = null;
+
+        if (!\is_array($duplicateFields)) {
+            $duplicateFields = [];
+            $fieldsetGroup = null;
+
+            foreach ($allFields as $field) {
+                if ($this->fieldHelper->isFieldsetStart((object) $field) && $field['allowDuplication'] && $field['leadStore']) {
```
Comment (fritzmg) [inline src/EventListener/LeadsListener.php:107]:
```suggestion
                if ($this->fieldHelper->isFieldsetStart((object) $field) && $field['allowDuplication'] && $field['leadStore'] ?? null) {
```

`leadStore` might not exist if leads is not installed, I think

**Review by fritzmg — CHANGES_REQUESTED** [review]

File: src/EventListener/LeadsListener.php (hunk 1 above)
Comment (qzminski) [reply src/EventListener/LeadsListener.php:107]: If the leads is not installed, then the hook should not be executed either. Or am I missing something?

Comment (fritzmg) [reply src/EventListener/LeadsListener.php:107]: Right 😁

**Comment by fritzmg** [comment]
Thank you :). Released as `1.4.0`.

**Comment by qzminski** [comment]
Thank you @fritzmg !!

**Pull request merged**